src/gui/modalMenu.cpp

```diff
@@ -286,12 +290,21 @@ bool GUIModalMenu::preprocessEvent(const SEvent &event)
 			return retval;
 		}
 	}
+#endif
 
+#ifdef HAVE_TOUCHSCREENGUI
 	if (event.EventType == EET_TOUCH_INPUT_EVENT) {
 		irr_ptr<GUIModalMenu> holder;
 		holder.grab(this); // keep this alive until return (it might be dropped downstream [?])
 
-		switch ((int)event.TouchInput.touchedCount) {
+		if (event.TouchInput.Event == ETIE_PRESSED_DOWN) {
+			m_touched_count++;
+		}
+		int old_touched_count = m_touched_count;
+		if (m_touched_count > 0 && event.TouchInput.Event == ETIE_LEFT_UP) {
+			m_touched_count--;
+		}
```

Comment:
Please add support for the touched count to IrrlichtMt instead of this.

Reply:
Done: minetest/irrlicht#60